### ISSUE TITLE
DIE Illegal division by zero at /root/ViPullStatistics.pl line 146.

### ISSUE THREAD

**Issue by kamilnew** [sexibytes/sexigraf#171]
vCSA 6.7 Update 1
[2018/10/23 11:55:35] /root/ViPullStatistics.pl line274 - [INFO] Processing vCenter x.x.x.x cluster xxx datastores in datacenter xxx
[2018/10/23 11:55:36] /root/ViPullStatistics.pl line146 - DIE Illegal division by zero at /root/ViPullStatistics.pl line 146.

got this error in log and graphs have "holes"

![bledygraf1](https://user-images.githubusercontent.com/29244326/47412394-f06dcb80-d76b-11e8-9a6a-ef1a364a33c4.PNG)
![bledygraf2](https://user-images.githubusercontent.com/29244326/47412395-f19ef880-d76b-11e8-8cbb-6cad34653a89.PNG)

**Comment by rschitz** [comment]
This will be fix in the next update, could you try this one and let me know https://github.com/sexibytes/sexigraf/blob/dev/root/ViPullStatistics.pl

**Comment by kamilnew** [comment]
tried it but it doesnt work - got this in log file

2018/10/24 11:25:10] /usr/local/share/perl/5.20.2/Net/Graphite.pm line120 - WARN Use of uninitialized value $hash in concatenation (.) or string at /usr/local/share/perl/5.20.2/Net/Graphite.pm line 120.
[2018/10/24 11:25:10] /usr/local/share/perl/5.20.2/Net/Graphite.pm line120 - WARN Use of uninitialized value $hash in concatenation (.) or string at /usr/local/share/perl/5.20.2/Net/Graphite.pm line 120.
[2018/10/24 11:25:10] /root/ViPullStatistics.pl line1214 - DIE Can't call method "val" on an undefined value at /root/ViPullStatistics.pl line 1214.

**Comment by rschitz** [comment]
try the last version

**Comment by kamilnew** [comment]
now i got this errors in log:
2018/10/24 17:00:11] /root/ViPullStatistics.pl line1219 - WARN Use of uninitialized value $StandaloneResourceVMHostName in string eq at /root/ViPullStatistics.pl line 1219.
[2018/10/24 17:00:11] /root/ViPullStatistics.pl line1226 - DIE Can't call method "val" on an undefined value at /root/ViPullStatistics.pl line 1226.

virtual machines stats are working now 
datastores latency and iops dont work at all

**Comment by rschitz** [comment]
is this one better? https://github.com/sexibytes/sexigraf/blob/069be584ff973b73d093fd5a8dacf1176185de4d/root/ViPullStatistics.pl

**Comment by kamilnew** [comment]
now there are no errors in logs but still no datastore data - i think there are datapoints but they are null

**Comment by kamilnew** [comment]
can you show me how manually gather information about datastores to check if this values are null because of problem in my environment?

btw. i wish my enterprise products have this quality of support :)

**Comment by rschitz** [comment]
Thanks for your support.
when you said "no datastore data" you talk about local datastores or shared one ? did you ever get data for those?

**Comment by kamilnew** [comment]
shared one`s - those from lower screenshot - they were partialy working when i was taking screenshot on older versoin of ViPullStatistics.pl  yesterday 
i check historic data on sexigraf and i lost iops data at 27th of september , at the same time "holes" in datastore latency appear - this was during vsphere/vsan reconfiguration and that is why i think it may be problem in my environment - try to find anything in logs

**Comment by rschitz** [comment]
holes generally means slow vcenter or too something that takes more than 5min to complete, by sliding pointer on the graphs you should see points but not enough to have lines.
check the Pull Exec Time dashboard to confirm

**Comment by kamilnew** [comment]
pull exec look ok - ill try to pull data manually to see what is misiing - im sure that is problem with my environemt - about 40% of datastores have data and 60% dont - thanks for help
![pullexec](https://user-images.githubusercontent.com/29244326/47545132-d8708600-d8ea-11e8-9127-7b7a1420df34.PNG)

**Comment by kamilnew** [comment]
problem solved - i disabled storage i/o control on several datastores and thats why there was no data

**Comment by kamilnew** [comment]
yes it is enabled by default  but i disabled it on purpose month ago - don`t even remember why :) thanks for help

**Comment by rschitz** [comment]
you're welcome, thanks for you support and feedback!
